src/document.rs

```diff
@@ -1,27 +1,204 @@
+    /// the result.
+    pub return_new: Option<bool>,
+    /// Additionally return the complete old document under the attribute old in
+    /// the result.
+    pub return_old: Option<bool>,
+    /// If set to true, an empty object will be returned as response.
+    /// No meta-data will be returned for the replaced document.
+    /// This option can be used to save some network traffic.
+    pub silent: Option<bool>,
+    /// You can conditionally replace a document based on a target revision id
+    /// by using the if-match HTTP header.
+    pub if_match: Option<String>,
+}
+/// Options for document reading.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub enum DocumentReadOptions {
+    /// If the “If-None-Match” header is given, then it must contain exactly one
+    /// Etag. The document is returned, if it has a different revision than
+    /// the given Etag. Otherwise an HTTP 304 is returned.
+    IfNoneMatch(String),
+    ///  If the “If-Match” header is given, then it must contain exactly one
+    /// Etag. The document is returned, if it has the same revision as the
+    /// given Etag. Otherwise a HTTP 412 is returned.
+    IfMatch(String),
+}
+/// Options for document removes,
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct DocumentRemoveOptions {
+    /// Wait until document has been synced to disk.
+    pub wait_for_sync: Option<bool>,
+    /// Additionally return the complete old document under the attribute old in
+    /// the result.
+    pub return_old: Option<bool>,
+    /// If set to true, an empty object will be returned as response.
+    /// No meta-data will be returned for the created document.
+    /// This option can be used to save some network traffic.
+    pub silent: Option<bool>,
+    /// You can conditionally replace a document based on a target revision id
+    /// by using the if-match HTTP header.
+    pub if_match: Option<String>,
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DocumentHeader {
     #[serde(skip_serializing_if = "String::is_empty")]
-    _id: String,
-
+    pub _id: String,
     #[serde(skip_serializing_if = "String::is_empty")]
-    _key: String,
-
+    pub _key: String,
     #[serde(skip_serializing_if = "String::is_empty")]
-    _rev: String,
+    pub _rev: String,
+}
 
+/// Standard Response when having CRUD operation on document
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DocumentResponse<T> {
+    #[serde(flatten)]
+    /// May contain the { _key : String, _id : String, _rev:String } of the
+    /// document
+    pub header: Option<DocumentHeader>,
+    /// May contain the document after being created/replace/updated
+    pub new: Option<T>,
+    #[serde(rename = "_oldRev")]
+    /// May contain the old revision of the document after update/replace
+    pub _old_rev: Option<String>,
+    /// May contain the old the document after update/replace/remove
+    pub old: Option<T>,
```

Comment:
Why all fields are wrapped in `Option`? I don't think the server can returned a reponse with arbitrary combination of the 4 fiedls. If you would like to divide into new docuemnt and updated document, use `Enum` instead. And use a constom deserialization, see `Response` as example.